staticd/static_nb_config.c

```diff
@@ -169,9 +169,15 @@ static bool static_nexthop_create(struct nb_cb_create_args *args)
 		if (iter.blackhole && iter.count > 1) {
 			snprintf(
 				args->errmsg, args->errmsg_len,
-				"Route can not have blackhole and non-blackhole nexthops simultaneously");
+				"Route cannot have blackhole and non-blackhole nexthops simultaneously");
```

Comment:
So on init, zebra sends back the multpath num it is currently running in `struct zclient_capabilities`.ecmp as part of the init.  In static_zebra.c look at the static_zebra_capabilities() function to read it.  Staticd should use this value to limit what it is sending to zebra instead of MULTIPATH_NUM because an operator can further limit the run of zebra to a lower ecmp value than what the compile was.

Reply:
ok, I've revised to capture the number zebra advertises and use that instead.